src/posts/posts.controller.spec.ts

```diff
@@ -58,10 +58,10 @@ describe('PostsController', () => {
   });
 
   describe('flagPost function', () => {
-    it('should call service.flagpost with passed parameters', () => {
+    it('should call service.flagpost with dto & postid', async () => {
```

Comment:
I don't understand why do we need this test ?

Reply:
in order to fulfill the one user request, the request goes through multiple components each has single responsibility (controller for collecting, validating and sometime transforming request data, then pass its result to the service, then the service does its job and pass it to the next component and so on...

this test ensures that the controller passes the expected user params and body to the next component (the service) so that in the future if someone changed the output of the controller this test should fail to remind him that the service might also needs some changes to accept the new shape of the data.